keps/sig-storage/1698-generic-inline-volumes/README.md

```diff
@@ -0,0 +1,636 @@
+[`FSType`](https://github.com/kubernetes/kubernetes/blob/1fb0dd4ec5134014e466509163152112626d52c3/pkg/apis/policy/types.go#L278-L309)
+it will be possible to limit the usage of this volume source via the
+[PodSecurityPolicy
+(PSP)](https://kubernetes.io/docs/concepts/policy/pod-security-policy/#volumes-and-file-systems).
+
+The normal namespace quota for PVCs in a namespace still applies, so
+even if users are allowed to use this new mechanism, they cannot use
+it to circumvent other policies.
+
+## Example
+
+Here is a full example for a higher-level object that uses ephemeral
+inline volumes:
+
+```
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: fluentd-elasticsearch
+  namespace: kube-system
+spec:
+  selector:
+    matchLabels:
+      name: fluentd-elasticsearch
+  template:
+    metadata:
+      labels:
+        name: fluentd-elasticsearch
+    spec:
+      containers:
+      - name: fluentd-elasticsearch
+        image: quay.io/fluentd_elasticsearch/fluentd:v2.5.2
+        volumeMounts:
+        - name: varlog
+          mountPath: /var/log
+        - name: scratch
+          mountPath: /scratch
+      volumes:
+      - name: varlog
+        hostPath:
+          path: /var/log
+      - name: scratch
+        ephemeral:
+          metadata:
+            labels:
+              type: fluentd-elasticsearch-volume
+          spec:
+            accessModes: [ "ReadWriteOnce" ]
+            storageClassName: "scratch-storage-class"
+            resources:
+              requests:
+                storage: 1Gi
+```
+
+The DaemonSet controller will create pods with names like
+`fluentd-elasticsearch-b96sd` and the new controller will then add a
+PVC called `fluentd-elasticsearch-b96sd-scratch` for that pod.
+
+## Design Details
+
+### PVC meta data
+
+The namespace is the same as for the pod and the name is the
+concatenation of pod name and the `Volume.Name`. This is guaranteed to
+be unique for the pod and each volume in that pod. Care must be taken
+by the user to not exceed the length limit for object names, otherwise
+volumes cannot be created.
+
+Other meta data is copied from the `VolumeClaimTemplate.ObjectMeta`.
+
+### Preventing accidental collision with existing PVCs
+
+The controller will only create missing PVCs. It does not need to
+delete (handled by garbage collection) nor update (`PodSpec.Volumes`
+is immutable) existing PVCs. Therefore there is no risk that the
+controller will accidentally modify unrelated PVCs.
+
+The volume scheduling library and kubelet must check that a PVC is
+owned by the pod before using it. Otherwise they must ignored it. This
+ensures that a volume is not used accidentally for a pod in case of a
+name conflict.
+
+### Pod events
+
+Errors that occur while creating the PVC will be reported as events
+for the pod and thus be visible to the user. This includes quota
+violations and name collisions.
+
+### Feature gate
+
+The `GenericInlineVolumes` feature gate controls whether:
```

Comment:
Should this be called GenericEphemeralVolume now?

Reply:
Okay. Do I now also need to rename the KEP to match?

Reply:
Yes

Reply:
I don't agree (see my other comment), but if that's what it takes to get the KEP merged... done.